Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift

```diff
@@ -60,41 +72,38 @@ public struct BPKSearchInputSummary: View {
                 .accessibilityHidden(true)
             TextField(placeholder, text: $text, prompt: placeholderView)
                 .font(style: .bodyDefault)
-                .foregroundColor(state.textColor)
-                .disabled(state.isDisabled)
+                .foregroundColor(.textPrimaryColor)
+                .disabled(readOnly)
                 .focused($focused)
                 .accessibilityAddTraits(.isSearchField)
-                .accessibilityLabel(placeholder)
-                .accessibilityIdentifier("search_field")
             accessory
         }
         .frame(maxWidth: .infinity, minHeight: 48.0)
         .padding(.horizontal, BPKSpacing.base)
         .background(.surfaceDefaultColor)
-        .clipShape(RoundedRectangle(cornerRadius: .sm))
-        .outline(focused ? .textLinkColor : state.borderColor, cornerRadius: .sm, lineWidth: focused ? 2.0 : 1.0)
+        .clipShape(RoundedRectangle(cornerRadius: .md))
+        .outline(
+            isBorderHighlighted ? .textLinkColor : .lineColor,
+            cornerRadius: .md,
+            lineWidth: isBorderHighlighted ? 2.0 : 1.0
+        )
         .if(!BPKFont.enableDynamicType, transform: {
             $0.sizeCategory(.large)
         })
     }
     
     @ViewBuilder
     private var accessory: some View {
-        if let icon = state.icon {
-            if case let .clear(accessibilityLabel, action) = state {
-                Button(action: action) {
-                    BPKIconView(icon.icon)
-                        .foregroundColor(icon.color)
-                }
-                .accessibilityElement(children: .ignore)
-                .accessibilityLabel(accessibilityLabel)
-                .accessibilityAddTraits(.isButton)
-                .opacity(text.isEmpty ? 0.0 : 1.0)
-            } else {
-                BPKIconView(icon.icon)
-                    .foregroundColor(icon.color)
-                    .accessibilityHidden(true)
+        if let clearAction {
+            let clearIcon = BPKSearchInputSummary.Icon(icon: .closeCircle, color: .textSecondaryColor)
+            Button(action: clearAction.action) {
+                BPKIconView(clearIcon.icon)
```

Comment:
I think we can just do the definition of this here, and remove the `BPKSearchInputSummary.Icon` struct altogether, it makes no longer sense

```swift
BPKIconView(.closeCircle)
    .foregourndColor(.textSecondaryColor)

Reply:
We still need this struct for a prefix, but yes, here we can simplify it and not use this struct for a clear action